.github/scripts/findUnusedStyles.ts

```diff
@@ -0,0 +1,462 @@
+                console.warn(`Warning: Could not read style file ${file}:`, error);
+            }
+        }
+    }
+
+    private extractStyleKeysFromFile(sourceFile: ts.SourceFile, file: string): void {
+        const visit = (node: ts.Node) => {
+            // For styles/index.ts, only process styles that are inside the main styles function
+            if (file === 'src/styles/index.ts') {
+                // Look for the main styles function (arrow function assigned to 'styles' variable)
+                if (ts.isVariableDeclaration(node) && ts.isIdentifier(node.name) && node.name.text === 'styles' && node.initializer && ts.isArrowFunction(node.initializer)) {
+                    // Process only the body of the styles function
+                    if (node.initializer.body) {
+                        let returnObject: ts.Expression | undefined;
+
+                        if (ts.isParenthesizedExpression(node.initializer.body)) {
+                            returnObject = node.initializer.body.expression;
+                        } else if (ts.isObjectLiteralExpression(node.initializer.body)) {
+                            returnObject = node.initializer.body;
+                        } else if (ts.isSatisfiesExpression(node.initializer.body)) {
+                            const satisfiesExpr = node.initializer.body.expression;
+
+                            if (ts.isParenthesizedExpression(satisfiesExpr)) {
+                                returnObject = satisfiesExpr.expression;
+                            } else {
+                                returnObject = satisfiesExpr;
+                            }
+                        }
+
+                        if (returnObject && ts.isObjectLiteralExpression(returnObject)) {
+                            // Process each property in the returned object
+                            this.processObjectProperties(returnObject, sourceFile, file);
+
+                            // After processing style definitions, check for spread patterns within this styles file
+                            const fileContent = sourceFile.getFullText();
+                            this.checkSpreadPatternsInStylesFile(fileContent);
+                        }
+                    }
+                    return; // Don't continue traversing for styles function
+                }
+
+                // Skip processing top-level constants and other functions for styles/index.ts
+                if (ts.isVariableDeclaration(node) || ts.isFunctionDeclaration(node)) {
+                    return;
+                }
+            } else {
+                // For other files, process all style-like nodes normally
+                this.processStyleNode(node, sourceFile, file);
+            }
+
+            ts.forEachChild(node, visit);
+        };
+
+        visit(sourceFile);
+    }
+
+    private processObjectProperties(objectLiteral: ts.ObjectLiteralExpression, sourceFile: ts.SourceFile, file: string): void {
```

Comment:
`processObjectProperties` isn't a very expressive function name. Let's consider a better name and/or add a JSDoc comment for this function